File: app/src/main/java/mb00/android/codehub/ui/user/viewmodel/UserOverviewViewModel.java
package mb00.android.codehub.ui.user.viewmodel;

import mb00.android.codehub.ui.user.interactor.UserInteractor;
import mb00.android.codehub.ui.user.router.UserRouter;


public class UserOverviewViewModel extends UserViewModel {

    public UserOverviewViewModel(UserInteractor interactor, UserRouter router) {
        super(interactor, router);
    }

}

File: app/src/main/java/mb00/android/codehub/ui/user/viewmodel/UserOverviewViewModel.kt
package mb00.android.codehub.ui.user.viewmodel

import io.reactivex.Single
import mb00.android.codehub.api.model.User
import mb00.android.codehub.ui.user.interactor.UserInteractor
import mb00.android.codehub.ui.user.router.UserRouter


class UserOverviewViewModel(private val interactor: UserInteractor, private val router: UserRouter)
    : UserViewModel(interactor, router) {

    fun loadUserOverview(header: String, user: String): Single<User> {
        return interactor.loadUserOverview(header, user)
    }

}
